agent/api/task/task.go

```diff
@@ -696,13 +707,33 @@ func (task *Task) BuildCNIConfig() (*ecscni.Config, error) {
 			break
 		}
 	}
-
 	// If there is ipv6 assigned to eni then set it
 	if len(eni.IPV6Addresses) > 0 {
 		cfg.ENIIPV6Address = eni.IPV6Addresses[0].Address
 	}
+}
+
+// convertAppMeshToCNIConfig converts input app mesh config into cni config
+func convertAppMeshToCNIConfig(appMesh *apiappmesh.AppMesh, cfg *ecscni.Config) {
+	cfg.AppMeshCNIEnabled = appMesh.AppMeshEnabled
+	cfg.IgnoredUID = appMesh.IgnoredUID
+	cfg.IgnoredGID = appMesh.IgnoredGID
+	cfg.ProxyIngressPort = appMesh.ProxyIngressPort
+	cfg.ProxyEgressPort = appMesh.ProxyEgressPort
+	cfg.AppPorts = appMesh.AppPorts
+
+	var egressIgnoredIPs []net.IP
+	for _, ip := range appMesh.EgressIgnoredIPs {
+		ip := net.ParseIP(ip)
```

Comment:
Do you know what validation that ECS service side is doing? Do they validate whether the ip address is a valid ipv4/ipv6 address and how do they handle heading or trailing spaces?

Reply:
since we are not validating parameters, this will be removed